helmfile/charts/networkpolicy-service/templates/harbor-np.yaml

```diff
@@ -0,0 +1,609 @@
+      component: chartmuseum
+  ingress:
+    - from:
+        - podSelector:
+            matchLabels:
+              component: core
+      ports:
+        - port: 9999
+  egress:
+{{- if eq .Values.harbor.redis.type "internal" }}
+    - to:
+        - podSelector:
+            matchLabels:
+              component: redis
+      ports:
+        - port: 6379
+{{- else if eq .Values.harbor.redis.type "external" }}
+  {{- range .Values.harbor.redis.external }}
+    - to:
+      {{- if .egressNSMatchLabels }}
+        - namespaceSelector:
+            matchLabels:
+              {{- range $key, $value := .egressNSMatchLabels }}
+              {{ $key | quote }}: {{ $value | quote }}
+              {{- end }}
+          {{- if .egressNSPodMatchLabels }}
+          podSelector:
+            matchLabels:
+              {{- range $key, $value := .egressNSPodMatchLabels }}
+              {{ $key | quote }}: {{ $value | quote }}
+              {{- end }}
+          {{- end }}
+      {{- end }}
+      ports:
+    {{- range $port := .ports }}
+        - port: {{ $port }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+    - to:
+{{- range $IP := .Values.harbor.chartmuseum.httpsIPs }}
+        - ipBlock:
+            cidr: {{ $IP }}
+{{- end }}
+      ports:
+        - protocol: TCP
+          port: 443
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-harbor-core
+  namespace: harbor
+spec:
+  policyTypes:
+    - Ingress
+    - Egress
+  podSelector:
+    matchLabels:
+      component: "core"
+  ingress:
+    - ports:
+        - port: 8080
+          protocol: TCP
+  egress:
+    - to:
+{{- range $IP := .Values.harbor.core.httpsIPs }}
+        - ipBlock:
+            cidr: {{ $IP }}
+{{- end }}
+      ports:
+        - protocol: TCP
+          port: 443
+    - to:
+        - podSelector:
+            matchLabels:
+              component: chartmuseum
+      ports:
+        - port: 9999
+    - to:
+        - podSelector:
+            matchLabels:
+              component: registry
+      ports:
+        - port: 5000
+{{- if eq .Values.harbor.redis.type "internal" }}
+    - to:
+        - podSelector:
+            matchLabels:
+              component: redis
+      ports:
+        - port: 6379
+{{- else if eq .Values.harbor.redis.type "external" }}
+  {{- range .Values.harbor.redis.external }}
+    - to:
+      {{- if .egressNSMatchLabels }}
+        - namespaceSelector:
+            matchLabels:
+              {{- range $key, $value := .egressNSMatchLabels }}
+              {{ $key | quote }}: {{ $value | quote }}
+              {{- end }}
+          {{- if .egressNSPodMatchLabels }}
+          podSelector:
+            matchLabels:
+              {{- range $key, $value := .egressNSPodMatchLabels }}
+              {{ $key | quote }}: {{ $value | quote }}
+              {{- end }}
+          {{- end }}
+      {{- end }}
+      ports:
+    {{- range $port := .ports }}
+        - port: {{ $port }}
+    {{- end }}
+  {{- end }}
```

Comment:
I know we have not used it a lot. But is it possible to use some sort of template for all of these redis and database sections that we repeat several times in the file? So that you just have to write it once and then add a reference to it.

Reply:
That would have been nice to have done yes :slightly_smiling_face: